features/0434-outofband/README.md

```diff
@@ -0,0 +1,465 @@
+# Aries RFC 0434: Out of Band Protocols
+
+- Authors: [Ryan West](ryan.west@sovrin.org), [Daniel Bluhm](daniel.bluhm@sovrin.org), Matthew Hailstone, [Stephen Curran](swcurran@cloudcompass.ca), [Sam Curren](sam@sovrin.org), [George Aristy](george.aristy@securekey.com)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2020-03-01
+- Status Note: This RFC extracts the `invitation` messages from the [DID Exchange](../../features/0023-did-exchange/README.md) protocol (and perhaps [Connection](../../features/0160-connection-protocol/README.md)), and replaces the combined `present_proof/1.0/request` combined with the `~service` decorator to define an ephemeral (connection-less) challenge.
+- Supersedes: Invitation Message in [0160-Connections](https://github.com/hyperledger/aries-rfcs/blob/9b0aaa39df7e8bd434126c4b33c097aae78d65bf/features/0160-connection-protocol/README.md#0-invitation-to-connect) and Invitation Message in [0023-DID-Exchange](https://github.com/hyperledger/aries-rfcs/blob/9b0aaa39df7e8bd434126c4b33c097aae78d65bf/features/0023-did-exchange/README.md#0-invitation-to-exchange).
+- Start Date: 2020-03-01
+- Tags: [feature](/tags.md#feature), [protocol](/tags.md#protocol)
+- Protocol Name: Out-of-Band Invitation, Out-of-Band Request
+- Version: 1.0, 1.0
+- URI: `http://didcomm.org/oob-invitation/%VER`, `http://didcomm.org/oob-request/%VER`
+
+## Summary
+
+Out Of Band protocols are used when you wish to engage with another agent and you don't have a DIDComm connection to use for the interaction.
+
+## Motivation
+
+The use of the `invitation` in the [Connection](../../features/0160-connection-protocol/README.md) and [DID Exchange](../../features/0023-did-exchange/README.md) protocols has been relatively successful, but has some shortcomings, as follows.
+
+### Connection Reuse
+
+A common pattern we have seen in the early days of Aries agents is a user with a browser getting to a point where a connection is needed between the website's (enterprise) agent and the user's mobile agent. A QR invitation is displayed, scanned and a protocol is executed to establish a connection. Life is good!
+
+However, with the current invitation processes, when the same user returns to the same page, the same process is executed (QR code, scan, etc.) and a new connection is created between the two agents. There is no way for the user's agent to say "Hey, I've already got a connection with you. Let's use that one!"
+
+We need the ability to reuse a connection.
+
+### Connection Establishment Versioning
+
+In the existing Connections and DID Exchange `invitation` handling, the _inviter_ dictates what connection establishment protocol all _invitee_'s will use. A more sustainable approach is for the _inviter_ to offer the _invitee_ a list of supported protocols and allow the _invitee_ to use one that it supports.
```

Comment:
It should be noted how failure should occur if the invitee does not support any of the provided protocols from the inviter.

Reply:
Interesting point - thanks. I've added what I think could be possible in this [commit](https://github.com/hyperledger/aries-rfcs/pull/434/commits/d65de658932403c6558f1413ec32d502b06297db).

Folks - please take a look and see what you think.

Reply:
I like it. I'm glad this was added. Good improvement.